### ISSUE TITLE
InvalidChallengeParameters on AGP 8 & Proguard

### ISSUE THREAD

**Issue by mikkoville** [Adyen/adyen-3ds2-android#53]
**Describe the bug**
We recently update our project to use AGP version 8
There were many changes how R8 is handled in this update:
https://stackoverflow.com/questions/76063416/issue-on-r8-minify

We are using currently version 2.2.11

When we try to show a native challenge using the sdk we are getting the following exception:
```
Caused by: com.adyen.threeds2.exception.InvalidInputException: Invalid challengeParameters.             
	at atd.z.a.a(Unknown Source:2)                                                                      
	at com.adyen.threeds2.internal.i.a(Unknown Source:14)                                               
	at com.adyen.threeds2.internal.i.doChallenge(Unknown Source:11)                                     
	at com.wolt.android.payment.payment_services.adyen.AdyenSdkImpl.doChallenge$lambda$4(AdyenSdk.kt:80)
```

This happens only when we obfuscate the code with R8 using minifyEnabled=true for the build

**To Reproduce**
- Use AGP version higher than 8.0.0
- Do not specify any special proguard-rules.pro
- Enter the flow with native challenge that builds ChallengeParameters

**Expected behavior**
- No crash

**Smartphone (please complete the following information):**
 - Device: [Android Pixel 6]

**Additional context**

NOTE I am able to make it not crash by entering the following in our `proguard-rules.pro`

```
-keep class com.adyen.** { *; }
-keep class org.bouncycastle.** { *; }
```

I chose to keep the above based on some previous issues in this sdk. I am not sure if this is enough for all cases though. Please confirm

**Comment by tkuntubayev** [comment]
Hi @mikkoville,
Thank you for providing all the details!

We will investigate the issue and come back soon if more details would be needed.

**Comment by tkuntubayev** [comment]
Adding these rules to `proguard-rules.pro` should prevent removing files that are used by SDK and should solve the issue:
```
-keep class org.bouncycastle.** { *; }
-dontwarn org.bouncycastle.**
```
These rules will be added to SDK with the next release.

Thank you for indication!

**Comment by Musthuakqha** [comment]
Good

**Comment by fdcb** [comment]
Hello, @Musthuakqha this issue has been fixed in our newest release 2.2.14.
Thank you for flagging it!
